java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -473,6 +473,39 @@ public void setrange(BaseClient client) {
         assertTrue(exception.getCause() instanceof RequestException);
     }
 
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void getrange(BaseClient client) {
+        String stringKey = UUID.randomUUID().toString();
+        String nonStringKey = UUID.randomUUID().toString();
+
+        assertEquals(OK, client.set(stringKey, "This is a string").get());
+        assertEquals("This", client.getrange(stringKey, 0, 3).get());
+        assertEquals("ing", client.getrange(stringKey, -3, -1).get());
+        assertEquals("This is a string", client.getrange(stringKey, 0, -1).get());
+
+        // out of range
+        assertEquals("string", client.getrange(stringKey, 10, 100).get());
+        assertEquals("This is a stri", client.getrange(stringKey, -200, -3).get());
+        assertEquals("", client.getrange(stringKey, 100, 200).get());
+
+        // incorrect range
+        assertEquals("", client.getrange(stringKey, -1, -3).get());
+
+        // a redis bug lol kek
+        assertEquals("T", client.getrange(stringKey, -200, -100).get());
```

Comment:
Is it a bug or is negative indices larger than string length actually equal to 0? That could be the case because the same happens in this case above assertEquals("This is a stri", client.getrange(stringKey, -200, -3).get());.

Reply:
Don't know why it does not happen for positive values 🤷

Reply:
Should return an empty string, that is a real bug